src/simulator/views/map/data/flat_map.py

```diff
@@ -53,8 +62,33 @@ def __init__(self, services: Services, data: NDArray[(Any, Any, Any), bool], par
                     break
 
         self.square_size = square_size
-        self.texture_w = int(self.logical_w * square_size)
-        self.texture_h = int(self.logical_h * square_size)
+
+        if block_size is None:
+            is_power_of_2 = lambda n: (n & (n-1) == 0) and n != 0
+            
+            # make block fit entire map
+            block_size = max(self.logical_w, self.logical_h)
+
+            # ensure block size is a power of 2 so that the
+            # size of its texture is a power of 2.
+            if not is_power_of_2(block_size):
```

Comment:
why make this into a lambda

Reply:
Thought I was going to use it more than once xD

Reply:
might wanna just get rid of that then no? code cleanup and all